src/routes/boot.ts

```diff
@@ -769,7 +769,7 @@ const resolveDynamicFunnelId = (userId: string): string => {
 
 // Fetches the funnel data from Freyja
 const getFunnel = async (req: FastifyRequest, res: FastifyReply) => {
-  const userId = req.userId || req.trackingId;
+  const userId = req?.userId || req?.trackingId;
```

Comment:
when is req undefined?

Reply:
Should never been undefined imo, but I'm trying to make every occurrence where we are picking userId safe.  I can revert it if needed. 

[Some context about this PR](https://dailydotdev.slack.com/archives/C08J2D9NKPU/p1744795560847689?thread_ts=1744726922.675339&cid=C08J2D9NKPU)

Reply:
even if it could happen we need to find the root cause. it seems like the issue is elsewhere

Reply:
IMO the issue is [here](https://github.com/dailydotdev/daily-api/pull/2769/files#diff-53d88b8a44fefbe970e714bd8cd0ed3245db56a6645de0d7f174a2a26fd524d9R783) and not in the req, but added those check to be sure 👀

Reply:
ah you're right. please remove this optional chaining then and keep the other one

Reply:
@idoshamun Done ✔️ 

Let's try only with this change, then if still not working I'll investigate more on the req @capJavert @sshanzel

Reply:
looks like in prod build indeed the line number `512:98` correlates to the changed line:
<img width="949" alt="image" src="https://github.com/user-attachments/assets/b0f57bc2-151f-47a9-b7a4-0673f612a93e" />

Reply:
so crash is on that part